.github/workflows/release.yml

```diff
@@ -10,19 +10,22 @@ permissions:
   contents: write
 
 env:
-  PREVIEW_TASK: publishMavenPublicationToSonatypeSnapshotRepository
-  PUBLISH_TASK: publishMavenPublicationToSonatypeRepository
+  PREVIEW_TASK: publishToSonatype
```

Comment:
QQ: why did we have to change the tasks names here? (I understand why we had to add one to the publish)

Reply:
I changed these to use the built-in tasks provided by the `gradle-nexus-publishing` plugin. We could still revert to the previous publish tasks provided by `maven-publish`.

However, I'd suggest we drop the maven-publish repository [configs](https://github.com/microsoft/kiota-java/blob/feat/automate-ossrh-release/components/abstractions/build.gradle#L105) and only keep the gradle-nexus-publishing ones defined in this PR so that we have only one set of repository configs. If this sounds good, I can make these changes to this PR.

This way we would only have one working set of publish tasks provided by gradle-nexus.

Reply:
The reason why we had multiple different tasks was because we needed to publish to another feed in the past. That feed went offline and if we can simplify the configuration that will be awesome.

Reply:
thanks for making the updates. I'm still unclear how the snapshot (preview) builds will get to the snapshot feed with the current configuration?

Reply:
The default URLs are [pre-configured](https://github.com/gradle-nexus/publish-plugin?tab=readme-ov-file#publishing-to-maven-central-via-sonatype-ossrh)

Reply:
Ah! I get you. Good question.
It's determined by the `-SNAPSHOT` suffix being present in the version. This is currently the [default](https://github.com/microsoft/kiota-java/blob/main/gradle.properties#L41). We remove the suffix for staging to happen.

Docs weren't too clear. Found this [test](https://github.com/gradle-nexus/publish-plugin/blob/master/src/compatTest/kotlin/io/github/gradlenexus/publishplugin/BaseNexusPublishPluginTests.kt#L90)

Reply:
Thank you for the additional information. In the current workflow, there's no place I could identify where we might be "removing" the snapshot suffix for central releases. Did I miss something?

Reply:
These were already being removed [here](https://github.com/microsoft/kiota-java/pull/1445/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R148-R162)

Reply:
thanks for clarifying, I had missed this subtility